tests/unit/setup-rendering-context-test.js

```diff
@@ -555,6 +561,295 @@ module('setupRenderingContext', function (hooks) {
     });
 
+    if (hasEmberVersion(3, 25)) {
+      // render tests for components in 3.25+ where we can use lexical scope
+      module('using render with a component in Ember >= 3.25', function () {
+        test('works with a template-only component', async function (assert) {
+          const name = 'Chris';
+
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+          const component = setComponentTemplate(template, templateOnly());
+          await render(component);
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris',
+            'has rendered content'
+          );
+        });
+
+        test('works with a glimmer component', async function (assert) {
+          const name = 'Chris';
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}} and my favorite color is {{this.favoriteColor}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+
+          class Foo extends GlimmerComponent {
+            favoriteColor = 'red';
+          }
+
+          setComponentTemplate(template, Foo);
+          await render(Foo);
+
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris and my favorite color is red',
+            'has rendered content'
+          );
+        });
+
+        test('works with a classic component', async function (assert) {
+          const name = 'Chris';
+          const template = precompileTemplate(
+            '<p>hello my name is {{name}} and my favorite color is {{this.favoriteColor}}</p>',
+            {
+              scope() {
+                return {
+                  name,
+                };
+              },
+            }
+          );
+
+          const Foo = Component.extend({
+            favoriteColor: 'red',
+          });
+
+          setComponentTemplate(template, Foo);
+          await render(Foo);
+
+          assert.equal(
+            this.element.textContent,
+            'hello my name is Chris and my favorite color is red',
+            'has rendered content'
+          );
+        });
+
+        test('components passed to render do *not* have access to the test context', async function (assert) {
+          const template = precompileTemplate(
+            'the value of foo is {{this.foo}}'
+          );
+
+          class Foo extends GlimmerComponent {
+            foo = 'foo';
+          }
+
+          const component = setComponentTemplate(template, Foo);
+          await render(component);
+
+          assert.equal(this.element.textContent, 'the value of foo is foo');
+
+          this.set('foo', 'bar');
```

Comment:
We'll need to tweak this when we fixup the `this.set` assertion, but the test overall is 👌